src/org/ods/orchestration/usecase/BitbucketTraceabilityUseCase.groovy

```diff
@@ -107,30 +107,54 @@ class BitbucketTraceabilityUseCase {
         boolean nextPage = true
         int nextPageStart = 0
         while (nextPage) {
-            Map commits = bitbucketService.getCommitsForIntegrationBranch(token, repo.repo, PAGE_LIMIT, nextPageStart)
+            Map pullRequests = bitbucketService.getMergedPullRequestsForIntegrationBranch(token, repo,
+                PAGE_LIMIT, nextPageStart)
+            if (pullRequests.isLastPage) {
+                nextPage = false
+            } else {
+                nextPageStart = pullRequests.nextPageStart
+            }
+
+            int pullRequestsSize = pullRequests.values.size()
+            for (def i = 0; i < pullRequestsSize; i++) {
+                def pullRequest = pullRequests.values[i]
+                records += processPullRequest(token, repo, pullRequest)
+            }
+        }
+        return records
+    }
+
+    @NonCPS
+    private List<Record> processPullRequest(String token, Map repo, Map pullRequest) {
+        def records = []
+        boolean nextPage = true
+        int nextPageStart = 0
+        while (nextPage) {
+            Map commits = bitbucketService.getCommmitsForPullRequest(token, repo.repo, pullRequest.id,
+                PAGE_LIMIT, nextPageStart)
             if (commits.isLastPage) {
                 nextPage = false
             } else {
                 nextPageStart = commits.nextPageStart
             }
-            records += processCommits(token, repo, commits)
+            records += processCommits(repo, commits, pullRequest)
         }
         return records
     }
 
     @NonCPS
-    private List<Record> processCommits(String token, Map repo, Map commits) {
+    private List<Record> processCommits(Map repo, Map commits, Map pullRequest) {
+        Map repoCommits = [:]
```

Comment:
Remove the map. Now this map does not avoid the duplicates between PR.

Reply:
done